### PR TITLE
Adding liking student submissions.

### PR DESCRIPTION
**Pull request by ZachBaird**

### CODE REVIEW

**Comment by ZachBaird** [comment]
Hey @KevinMulhern , I have yet to write some tests and button this up, but wanted to get this draft PR up to show you what I've written. If you have any suggestions like a clean way to sort on the server-side or something I missed, please let me know.

The results of this PR as-is include:

- A logged in user can click on a heart icon to like a submission. This will increment the votes and light up the heart for that submission. It cannot be clicked again to further vote on the submission nor unlike it.
- If the user visits a project page, all liked projects will already be lit up and cannot be liked further.
- If a user visits the project page and isn't authenticated, the heart icon and votes will not show up at all. The projects will still be sorted by most-liked.
- The submissions are sorted client-side atm based on # of likes.

**Review by KevinMulhern — COMMENTED** [review]
This is looking really good @ZachBaird 💪 very nice work!

I've left a few suggestions, they are all pretty minor. Let me know if you have any questions about any of them.

File: app/helpers/projects_helper.rb [hunk 1]
```diff
@@ -2,4 +2,13 @@ module ProjectsHelper
   def heart_icon_or_button_for(user, project)
     HeartIconPresenter.render(user: user, project: project, view: self)
   end
+
+  def project_submission_params
```
[the remark this answers is not in the document]
Comment (ZachBaird) [reply app/helpers/projects_helper.rb:6]: I actually was going to ask about this but forgot. I moved that code to the helper so I could reference it in both the likes and project_submission controllers. If it'd make more sense to toss it in a new project_submission_helper file instead, I'm totally down to make that. I just threw it in here for DRY.

File: app/javascript/components/project-submissions/containers/project-submissions-container.js [hunk 2]
```diff
@@ -85,6 +85,32 @@ const ProjectSubmissions = (props) => {
...
+        submission_id: submission.id,
+        is_liked_by_current_user: submission.is_liked_by_current_user
+      }
+    );
+
+    if (response.status === 200) {
+      const updatedSubmission = response.data;
+
+      setSubmissions(prevSubmissions => {
+        const newSubmissions = prevSubmissions.map((sub) => {
```
Comment (KevinMulhern) [inline app/javascript/components/project-submissions/containers/project-submissions-container.js:101]: Nitpick, can we change `sub` to `submissions` please? abbreviations are the devil 😆

**Review by KevinMulhern — COMMENTED** [review]
Really nice refactor @ZachBaird 💪  this is almost there

File: app/helpers/projects_helper.rb (hunk 1 above)
[the remark this answers is not in the document]
Comment (ZachBaird) [reply app/helpers/projects_helper.rb:6]: > it doesn't look like its being used in the likes controller? eyes

Lol you're right, not sure why I have it in there now

**Pull request closed without merging**